aunt-leahs-app/src/pages/AdminShiftDataPage.jsx

```diff
@@ -16,39 +17,52 @@ const AdminShiftDataPage = () => {
 			const response = await fetch('http://localhost:7071/api/Shifts', {
 				method: 'GET'
 			});
-
 			const shifts = await response.json();
-			
-			const shiftData = shifts.map(shift => {
+			const shiftData = shifts.map((shift) => {
 				const date = new Date(shift.startTime);
 				// The format of the date and time can be adjusted to the customer's needs
-				return ({
+				return {
 					id: shift.id,
 					firstName: shift.firstName,
 					lastName: shift.lastName,
 					date: date.toDateString(),
 					time: date.toTimeString(),
 					duration: shift.duration
-				});
+				};
 			});
-
 			setShifts(shiftData);
-		}
-		catch (error) {
-			console.log("Error fetching shift data: " + error);
+		} catch (error) {
+			console.log('Error fetching shift data: ' + error);
 		}
 	}
 
-	async function clearShifts() {
+	const exportData = () => {
+		const options = {
+			fieldSeparator: ',',
+			filename: 'shiftData',
+			quoteStrings: '"',
+			decimalSeparator: '.',
+			showLabels: true,
+			showTitle: true,
+			title: 'Volunteer Data',
+			useTextFile: false,
+			useBom: true,
+			useKeysAsHeaders: true
+			// headers: ['Column 1', 'Column 2', etc...] <-- Won't work with useKeysAsHeaders present!
+		};
+		const csvExporter = new ExportToCsv(options);
```

Comment:
This instantiates an exporter every time the button is clicked. While a user probably won't click this more than once, I think you should make a class variable and if it's not null, then create an instance.